Packet++/src/SipLayer.cpp

```diff
@@ -307,7 +307,7 @@ void SipRequestFirstLine::parseVersion()
 	verPos++;
 
 	int endOfVerPos = 0;
-	while (((verPos+endOfVerPos)[0] != '\r') && ((verPos+endOfVerPos)[0] != '\n'))
+	while (((verPos+endOfVerPos)[0] != '\r') && ((verPos+endOfVerPos)[0] != '\n') && ((verPos + endOfVerPos) < (char *) (m_SipRequest->m_Data + m_SipRequest->m_DataLen)))

```

Comment:
I suggest we switch the order and check `((verPos + endOfVerPos) < (char *) (m_SipRequest->m_Data + m_SipRequest->m_DataLen))` first (before we check for `\r` and `\n`). 

The reason is that if `verPos + endOfVerPos` is already at the end of the packet, then reading the byte in index zero: `(verPos+endOfVerPos)[0]` is a buffer overflow. I actually discovered that from one of the files you added as examples: `sip_layer_parse_version.bug`.

Reply:
I fixed that